mockafka/aiokafka/aiokafka_consumer.py

```diff
@@ -36,8 +36,15 @@ def message_to_record(message: Message, offset: int) -> ConsumerRecord[bytes, by
     key_str: Optional[str] = message.key()
     value_str: Optional[str] = message.value()
 
-    key = key_str.encode() if key_str is not None else None
-    value = value_str.encode() if value_str is not None else None
+    if not isinstance(key_str, bytes):
+        key = key_str.encode() if key_str is not None else None
+    else:
+        key = key_str
+
+    if not isinstance(value_str, bytes):
+        value = value_str.encode() if value_str is not None else None
+    else:
+        value = value_str
```

Comment:
This implies that the typing of `key_str` and `value_str` are incorrect -- perhaps they need to be `str | bytes | None`. I'd also suggest checking the typing of `Message` since I likely copied these values from there.

Reply:
I agree we want to always return bytes. I don't really understand why `ConsumerRecord` is generic rather than always offering `bytes`, though if as you suggest that's a setting then perhaps we should follow that? Either way I don't think we want to change that as part of this fix -- sticking with only bytes feels fine to me until we have a good reason to change.

My concern was the annotations of `key_str` and `value_str` on lines 36 & 37, which are effectively determined by the internal `Message` type's field types. That type currently implies that it can *only* contain strings (or `None`), which is why the previous code doesn't handle `bytes`. If it can legitimately be given `btytes` then the signature there should change and the annotations here updated similarly.